### PR TITLE
Origin/executor

### PR DESCRIPTION
**Pull request by gurujasko11**

### CODE REVIEW

File: Jimage/src/Executor.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.stream.Stream;
+
+/**
+ * Created by busz on 03.12.16.
+ */
```
Comment (slaby93) [inline Jimage/src/Executor.java]: remove this comment, it's useless

File: Jimage/src/Executor.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+public abstract class Executor {
+    Collection<Plugin> plugins;
+    Collection<Image> images;
+    Params parameters;
+
+    //laduje uzywajac input modules
```
Comment (slaby93) [inline Jimage/src/Executor.java]: remove polish comment!!!

File: Jimage/src/Executor.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            paths.forEach(filePath -> {
+                if (Files.isRegularFile(filePath)) {
+                    try {
+                        loadImage( filePath.toString() );
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+            });
+        } catch(Exception e){
```
Comment (slaby93) [inline Jimage/src/Executor.java]: I did not check if this works but I like this approach :)

File: Jimage/src/Executor.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        }
+    }
+
+    void addPlugin(Plugin p){
+        plugins.add(p);
+    }
+
+    abstract void process();
+
+    //getResultSet
```
Comment (slaby93) [inline Jimage/src/Executor.java]: remove

File: Jimage/src/Image.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+
+
+/**
+ * Created by busz on 03.12.16.
+ */
```
Comment (slaby93) [inline Jimage/src/Image.java]: remove

File: Jimage/src/Image.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+public class Image {
+    BufferedImage image ;
+
+    public Image(File f) throws IOException {
+        image = ImageIO.read( f );
+    }
+
+
```
Comment (slaby93) [inline Jimage/src/Image.java]: add getBufferedImage and consider immutable values :)

File: Jimage/src/JImageStreamToolkit.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by busz on 03.12.16.
+ */
+public class JImageStreamToolkit {
```
Comment (slaby93) [inline Jimage/src/JImageStreamToolkit.java]: I think this should be singleton

File: Jimage/src/JImageStreamToolkit.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+    static List<Plugin> plugins = new ArrayList<>();
```
Comment (slaby93) [inline Jimage/src/JImageStreamToolkit.java]: should this class contain plugins? I think executor should contain them.

File: Jimage/src/Params.java [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import java.util.HashMap;
+
+/**
+ * Created by busz on 03.12.16.
+ */
```
Comment (slaby93) [inline Jimage/src/Params.java]: remove

File: Jimage/src/Plugin.java [hunk 10]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Created by busz on 03.12.16.
+ */
```
Comment (slaby93) [inline Jimage/src/Plugin.java]: remove

File: Jimage/src/ResultImagesSet.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Created by busz on 03.12.16.
+ */
```
Comment (slaby93) [inline Jimage/src/ResultImagesSet.java]: remove

File: Jimage/src/Test2Plugin.java [hunk 12]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Created by janusz on 04.12.16.
+ */
```
Comment (slaby93) [inline Jimage/src/Test2Plugin.java]: remove

File: Jimage/src/Test2Plugin.java [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+//maluje prawa dolna cwiartke na czarno
```
Comment (slaby93) [inline Jimage/src/Test2Plugin.java]: this could be in some kind of javadoc to be used in IDE hints :)

File: Jimage/src/TestExecutor.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Vector;
+
+/**
+ * Created by janusz on 04.12.16.
+ */
```
Comment (slaby93) [inline Jimage/src/TestExecutor.java]: remove

File: Jimage/src/TestPlugin.java [hunk 15]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Created by janusz on 04.12.16.
+ */
```
Comment (slaby93) [inline Jimage/src/TestPlugin.java]: remove

File: Jimage/src/TestPlugin.java [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+//maluje lewa gorna cwiartke na czerwono
```
Comment (slaby93) [inline Jimage/src/TestPlugin.java]: same, go to javadoc

File: Jimage/src/TestPlugin.java [hunk 17]
```diff
@@ -0,0 +1,11 @@
...
+public class TestPlugin extends Plugin{
+    public void exec(Image image){
+        for(int i = 0; i < image.image.getHeight()/2;i++)
+            for(int j= 0;j < image.image.getWidth()/2; j++)
+                image.image.setRGB(j,i, 0xFF0000);
+    }
+}
```
Comment (slaby93) [inline Jimage/src/TestPlugin.java]: nice :)

**Comment by slaby93** [comment]
Where is any response and other people CR's ?!?!?!

File: Jimage/src/TestPlugin.java (hunk 17 above)
Comment (kmisztal) [inline Jimage/src/TestPlugin.java]: please check if it correct, because you probably may forget about alpha channel

**Comment by slaby93** [comment]
Any progress/comments?

**Pull request merged**